bundles/pubsub/pubsub_spi/src/pubsub_endpoint.c

```diff
@@ -181,7 +181,9 @@ celix_properties_t* pubsubEndpoint_createFromPublisherTrackerInfo(bundle_context
     }
 
     free(topic);
-    free(scopeFromFilter);
+    if (scope != NULL) {
```

Comment:
I left these as a free on NULL is not a problem. But I'm fine with those changes as well.